src/frontends/pytorch/src/op/index_add.cpp

```diff
@@ -0,0 +1,105 @@
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_index_add(const NodeContext& context) {
+    // aten::index_add(Tensor self, int dim, Tensor index, Tensor source, *, Scalar alpha=1) -> Tensor
+    // aten::index_add.out(Tensor self, int dim, Tensor index, Tensor source, *, Scalar alpha=1, Tensor(a!) out) ->
+    // Tensor(a!)
+    num_inputs_check(context, 5, 6);
+    auto input = context.get_input(0);
+    auto dim = context.get_input(1);
+    auto index = context.mark_node(std::make_shared<v0::Convert>(context.get_input(2), element::i32));
+    auto src = context.get_input(3);
+    auto alpha = context.get_input(4);
+    auto converted_alpha = context.mark_node(std::make_shared<v1::ConvertLike>(alpha, src));
+    auto alpha_src = context.mark_node(std::make_shared<v1::Multiply>(converted_alpha, src));
+    auto input_shape_rank = get_shape_rank(context, input);
+    auto const_one = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {1}));
+    auto const_one_0d = context.mark_node(v0::Constant::create(element::i32, Shape{}, {1}));
+    auto inp_rank = std::get<1>(input_shape_rank);
+    // ScatterElementsUpdate required that index, source and update have the same rank
+    // in aten::index_add index represents as 1d-array for specific dim and update may have different size
+    // from source in non-indexing axes
+    // slice src for having only relevant data
+    auto src_broadcast_shape = context.mark_node(std::make_shared<v3::Broadcast>(const_one, inp_rank));
+    auto src_broadcasted = context.mark_node(
+        std::make_shared<v3::Broadcast>(alpha_src, src_broadcast_shape, BroadcastType::BIDIRECTIONAL));
+    auto src_shape_rank = get_shape_rank(context, src_broadcasted);
+    auto const_zero = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {0}));
+    auto src_rank = std::get<1>(src_shape_rank);
+    auto slice_start = context.mark_node(std::make_shared<v3::Broadcast>(const_zero, inp_rank));
+    auto axes = get_node_axes_range(context, src_broadcasted);
+    auto const_inf =
+        context.mark_node(v0::Constant::create(element::i32, Shape{1}, {std::numeric_limits<int32_t>::max()}));
+    auto slice_end = context.mark_node(std::make_shared<v3::Broadcast>(const_inf, src_rank));
+    auto slice_step = context.mark_node(std::make_shared<v3::Broadcast>(const_one, src_rank));
+    auto dim_1d = context.mark_node(std::make_shared<v3::Broadcast>(dim, const_one, BroadcastType::BIDIRECTIONAL));
+    slice_end =
+        context.mark_node(std::make_shared<v12::ScatterElementsUpdate>(slice_end,
+                                                                       dim_1d,
+                                                                       const_one,
+                                                                       const_zero,
+                                                                       v12::ScatterElementsUpdate::Reduction::NONE));
```

Comment:
```suggestion
                                                                       const_zero));
```
None reduction is used by default

Reply:
applyed